cli/src/main/java/com/devonfw/tools/ide/tool/docker/Docker.java

```diff
@@ -33,6 +37,36 @@ public Docker(IdeContext context) {
     super(context, "docker", Set.of(Tag.DOCKER));
   }
 
+  @Override
+  public String getBinaryName() {
+    return detectContainerRuntime();
+  }
+
+  private String detectContainerRuntime() {
+    if (isCommandAvailable(this.getName())) {
+      return this.getName();
+    } else if (isCommandAvailable(PODMAN)) {
+      return PODMAN;
+    } else {
+      return this.getName();
+    }
```

Comment:
I assume this will be called multiple times. Shall we recompute the result every time or store the result once computed?
Maybe if we change the implementation so we do not trigger processes but just do a Path search, this will be "cheap" enough to recompute every time.
However, I do not expect a docker or podman installation to come and go on a daily basis (and surely when we find nothing, the variable would still be null and the recomputation should still take place, but once we found podman but no docker, I do not expect this to change within the running process - even in GUI dashboard).

Reply:
I've changed the isCommandAvailable logic to just check for the path in the PATH. So this can be run multiple times now.